### PR TITLE
Add automatic library generation

### PR DESCRIPTION
**Pull request by pablogs9**
This PR creates an automation for daily rebuild of the micro-ROS library, same approach as in https://github.com/micro-ROS/micro_ros_arduino

This way, users always can use latest micro-ROS features just by using the precompiled library.

Most relevant new features:
- RMW static memory pools https://github.com/micro-ROS/rmw-microxrcedds/pull/113
- micro-ROS external transports (built at app build time instead of included in `libmicroros.a`)
- micro-ROS epoch/time synchronization https://github.com/micro-ROS/rmw-microxrcedds/pull/119
- Bugfixes

@lukicdarkoo our Picos are in transit... is possible for you to tests this and let us know if everything works well at runtime? It should be just follow the Readme instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,15 +50,27 @@ docker run -it --rm -v /dev:/dev --privileged --net=host microros/micro-ros-agen
...
+If you want to compile it by yourself:
 
+<!-- 
+pushd extras/library_generation
+docker build . -t microros/micro_ros_arduino_builder:foxy
+popd
+ -->
+
+```bash
+docker pull microros/micro_ros_arduino_builder:foxy
```
Comment (lukicdarkoo) [inline README.md:69]:
```suggestion
```
This is not needed, the line below automatically pulls the image.

File: README.md [hunk 2]
```diff
@@ -50,15 +50,27 @@ docker run -it --rm -v /dev:/dev --privileged --net=host microros/micro-ros-agen
...
+docker run -it --rm -v $(pwd):/arduino_project microros/micro_ros_arduino_builder:foxy
```
Comment (lukicdarkoo) [inline README.md:70]: This command returns error:
```
cp: target '/arduino_project/libmicroros/include' is not a directory
```
if the `/libmicroros/include` directory doesn't exist.

Ideally, it should be handled by the Docker image or documented.

**Review by lukicdarkoo — CHANGES_REQUESTED** [review]
Nice! I can confirm it works.

A few minor comments.

Also, all files are now executable:
```
-rwxrwxr-x 1 lukic lukic   110 Mar 31 22:34 3rd-party-licenses.txt
-rwxrwxr-x 1 lukic lukic  7319 Mar 31 22:34 available_ros2_types
-rwxrwxr-x 1 lukic lukic  2788 Mar 31 22:34 built_packages
-rwxrwxr-x 1 lukic lukic   991 Mar 31 22:34 CMakeLists.txt
-rwxrwxr-x 1 lukic lukic  4738 Mar 31 22:34 CONTRIBUTING.md
drwxrwxr-x 3 lukic lukic  4096 Mar 31 22:34 extras
drwxrwxr-x 3 lukic lukic  4096 Mar 31 22:34 libmicroros
-rwxrwxr-x 1 lukic lukic 11357 Mar 31 22:34 LICENSE
-rwxrwxr-x 1 lukic lukic   961 Mar 31 22:34 NOTICE
-rwxrwxr-x 1 lukic lukic  1558 Mar 31 22:34 pico_micro_ros_example.c
-rwxrwxr-x 1 lukic lukic  1470 Mar 31 22:34 pico_uart_transport.c
-rwxrwxr-x 1 lukic lukic   581 Mar 31 22:34 pico_uart_transports.h
-rwxrwxr-x 1 lukic lukic  3192 Mar 31 22:34 README.md
```

File: README.md (hunk 1 above)
Comment (pablogs9) [reply README.md:69]: I'm going to keep it to ensure users to have the latest Dockerhub image, if an outdated local image exists, `docker run` does not update it.

File: README.md (hunk 2 above)
Comment (pablogs9) [reply README.md:69]: Solved: https://github.com/micro-ROS/micro_ros_raspberrypi_pico_sdk/blob/c88f949f3038f90ee7bbd60e1b92ed66282abce6/extras/library_generation/library_generation.sh#L47

**Comment by pablogs9** [comment]
Permissions solved, I'm merging to check if the workflow works.

**Pull request merged**